src/background/websocket-manager.ts

```diff
@@ -0,0 +1,244 @@
+  scriptHash: string,
+  store: Store
+) {
+  const historyResp = (await ws.request(
+    'blockchain.scripthash.get_history',
+    scriptHash
+  )) as History;
+  const historyDiff = selectHistoryDiff(historyResp)(store.getState());
+
+  const accountID = selectAccountIDByScriptHash(scriptHash, network)(store.getState());
+  const blindKeyGetter = getPrivateBlindKeyGetter(store, network);
+  const walletScripts = await getAllWalletScripts(store, network);
+
+  for (const { txID, height } of historyDiff.newTxs) {
+    const txFromElectrum = await fetchTx(ws, txID, height > 0 ? height : undefined);
+    const { unblindedTx } = await unblindTransaction(txFromElectrum, blindKeyGetter);
+    const displayTx = toDisplayTransaction(unblindedTx, walletScripts, networks[network]);
+
+    // add the new tx to the store
+    store.dispatch(addTx(accountID, displayTx, network));
+
+    // check if any utxos were spent in this tx
+    for (const input of unblindedTx.vin) {
+      const [utxoInStore, accountID] = selectUtxoByOutpoint(input, network)(store.getState());
+      if (utxoInStore) store.dispatch(deleteUtxo(accountID, input.txid, input.vout, network));
+    }
+
+    // check if we have any new utxos
+    const walletOutputs = unblindedTx.vout.filter((o) => {
+      if (isConfidentialOutput(o)) {
+        return isUnblindedOutput(o);
+      }
+
+      return walletScripts.includes(o.prevout.script.toString('hex'));
+    });
+
+    // add the new utxos to the store
+    const actions = walletOutputs.map((o) => addUtxo(accountID, o as UnblindedOutput, network));
+    actions.forEach(store.dispatch);
+  }
+
+  for (const { txID, height } of historyDiff.confirmedTxs) {
+    if (height <= 0) continue;
+    const blockHeaderHex = (await ws.request('blockchain.block.header', height)) as string;
+    const blockHeader = deserializeBlockHeader(blockHeaderHex);
+    const action = confirmTx(txID, blockHeader.timestamp, network);
+    store.dispatch(action);
+  }
+}
+
+// fetch a tx (and its prevouts) from websocket endpoint using the blockchain.transaction.get method
+async function fetchTx(ws: ElectrumWS, txID: string, blockheight?: number): Promise<TxInterface> {
+  const txHex = (await ws.request('blockchain.transaction.get', txID)) as string;
+  const transaction = Transaction.fromHex(txHex);
+
+  let blockHeader: BlockHeader | undefined;
+  if (blockheight) {
+    const blockHeaderHex = (await ws.request('blockchain.block.header', blockheight)) as string;
+    blockHeader = deserializeBlockHeader(blockHeaderHex);
+  }
+
+  const txInterface: TxInterface = {
+    txid: txID,
+    fee: 0,
+    status: {
+      confirmed: blockHeader !== undefined,
+      blockHeight: blockHeader?.height,
+      blockTime: blockHeader?.timestamp,
+    },
+    vin: [],
+    vout: [],
+  };
+
+  for (const input of transaction.ins) {
+    const prevoutTxID = Buffer.from(input.hash).reverse().toString('hex');
+    const vout = input.index;
+
+    if (!input.isPegin) {
```

Comment:
avoid wrapping in curlybraces?

```suggestion
    if (input.isPegin) return;
```